### PR TITLE
Chapter 5: Filling in the layout

### PR DESCRIPTION
**Pull request by baaohuii**

### CODE REVIEW

**Comment by baaohuii** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
conflict

**Comment by baaohuii** [comment]
ready

File: Gemfile [hunk 1]
```diff
@@ -4,8 +4,10 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 ruby "2.7.1"
 
 gem "bootsnap", ">= 1.4.2", require: false
+gem 'bootstrap-sass', '3.4.1'
```
Comment (manhnt-0960) [inline Gemfile]: use `"`

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+.center {
+text-align: center;
+}
+.center h1 {
+margin-bottom: 10px;
+}
+
+/* header */
+#logo {
+    float: left;
```
Comment (manhnt-0960) [inline app/assets/stylesheets/custom.scss]: indent 2 spaces

**Comment by baaohuii** [comment]
ready

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -13,3 +13,5 @@
  *= require_tree .
  *= require_self
  */
+@import "bootstrap-sprockets";
+@import "bootstrap";
```
Comment (saobv-2223) [inline app/assets/stylesheets/application.scss]: using '

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -0,0 +1,60 @@
+@import "bootstrap-sprockets";
+@import "bootstrap";
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: using '

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+ text-align: center;
+}
+
+h2 {
+ font-size: 1.2em;
+ letter-spacing: -1px;
+ margin-bottom: 30px;
+ text-align: center;
+ font-weight: normal;
+ color: #777;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: image

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  <%= link_to "sample app", '#', id: "logo" %>
+  <nav>
+   <ul class="nav navbar-nav navbar-right">
+    <li><%= link_to "Home",'#' %></li>
+    <li><%= link_to "Help",'#' %></li>
+    <li><%= link_to "Log in", '#' %></li>
+   </ul>
+  </nav>
+ </div>
+</header>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: image

File: app/views/layouts/_header.html.erb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+ <div class="container">
+  <%= link_to "sample app", '#', id: "logo" %>
+  <nav>
+   <ul class="nav navbar-nav navbar-right">
+    <li><%= link_to "Home",'#' %></li>
+    <li><%= link_to "Help",'#' %></li>
+    <li><%= link_to "Log in", '#' %></li>
```
Comment (saobv-2223) [inline app/views/layouts/_header.html.erb]: using "

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -10,6 +10,9 @@
   </head>
 
   <body>
+   <%= render "layouts/header" %>
+   <div class="container">
     <%= yield %>
+   </div>
   </body>
-</html>
+ </html>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]: image

**Comment by baaohuii** [comment]
ready

File: app/models/user.rb [hunk 9]
```diff
@@ -0,0 +1,2 @@
+class User < ApplicationRecord
+end
```
Comment (saobv-2223) [inline app/models/user.rb]: this file was use in chapter 6, so delete this file

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -10,6 +10,10 @@
...
+
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]: image

File: db/migrate/20200824080753_create_users.rb [hunk 11]
```diff
@@ -0,0 +1,10 @@
+class CreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :email
+
+      t.timestamps
+    end
+  end
+end
```
Comment (saobv-2223) [inline db/migrate/20200824080753_create_users.rb]: this file was use in chapter 6, so delete this file

**Comment by baaohuii** [comment]
ready

File: app/views/layouts/application.html.erb (hunk 10 above)
Comment (baaohuii) [reply app/views/layouts/application.html.erb]: done

File: db/migrate/20200824080753_create_users.rb (hunk 11 above)
Comment (baaohuii) [reply db/migrate/20200824080753_create_users.rb]: done

File: app/models/user.rb (hunk 9 above)
Comment (baaohuii) [reply app/models/user.rb]: done

**Review by saobv-2223 — COMMENTED** [review]
your pull is having 2 commits, merge to 1 commit, please

**Comment by baaohuii** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
Run command to merge your commit
`git reset --soft fd875f9b17bf86199c70e502e9d9181c06c41b00`
`git commit --amend`, now change your commit description, then press `Ctrl X -> Y -> Enter`
`git push origin -f init_project`

**Comment by baaohuii** [comment]
ready

File: Gemfile.lock [hunk 12]
```diff
@@ -118,6 +123,10 @@ GEM
       dry-initializer (~> 3.0)
       dry-schema (~> 1.5)
     erubi (1.9.0)
+<<<<<<< HEAD
+=======
+    execjs (2.7.0)
+>>>>>>> Init Project
```
Comment (chungpv-1008) [inline Gemfile.lock]: fix conflict

File: Gemfile.lock [hunk 13]
```diff
@@ -261,6 +270,10 @@ PLATFORMS
 
 DEPENDENCIES
   bootsnap (>= 1.4.2)
+<<<<<<< HEAD
+=======
+  bootstrap-sass (= 3.4.1)
+>>>>>>> Init Project
```
Comment (chungpv-1008) [inline Gemfile.lock]: similar

File: app/assets/stylesheets/custom.scss [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+/* header */
+#logo {
+ float: left;
+ margin-right: 10px;
+ font-size: 1.7em;
+ color: $logo;
+ text-transform: uppercase;
+ letter-spacing: -1px;
+ padding-top: 9px;
+ font-weight: bold;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  padding-top: 60px;
}

section {
  overflow: auto;
}

textarea {
  resize: vertical;
}

.center {
  text-align: center;
}

.center h1 {
  margin-bottom: 10px;
}

/* header */
#logo {
   float: left;
   margin-right: 10px;
   font-size: 1.7em;
   color: $logo;
   text-transform: uppercase;
   letter-spacing: -1px;
   padding-top: 9px;
   font-weight: bold;
```

**Comment by baaohuii** [comment]
ready

**Comment by manhnt-0960** [comment]
merged

**Pull request merged**